api/tests/unit/app_analytics/test_tasks.py

```diff
@@ -0,0 +1,322 @@
+    # since we started at 7:10:47, the first bucket should have 5 events
+    assert buckets[0].total_count == 5
+
+    # the rest of them should have 15 events
+    assert buckets[1].created_at.isoformat() == "2023-01-19T07:15:00+00:00"
+    assert buckets[1].total_count == 15
+
+    assert buckets[2].created_at.isoformat() == "2023-01-19T07:30:00+00:00"
+    assert buckets[2].total_count == 15
+
+    assert buckets[3].created_at.isoformat() == "2023-01-19T07:45:00+00:00"
+    assert buckets[3].total_count == 15
+
+    # Now, let's move forward 1hr and run this again
+    freezer.move_to(timezone.now() + timezone.timedelta(hours=1))
+    populate_api_usage_bucket(bucket_size, run_every=60)
+
+    # Then - we should have another four buckets created by the second run
+    buckets = APIUsageBucket.objects.filter(bucket_size=15).order_by("created_at").all()
+    assert len(buckets) == 8
+
+    # with 15 events each
+    assert buckets[4].created_at.isoformat() == "2023-01-19T08:00:00+00:00"
+    assert buckets[4].total_count == 15
+
+    assert buckets[5].created_at.isoformat() == "2023-01-19T08:15:00+00:00"
+    assert buckets[5].total_count == 15
+
+    assert buckets[6].created_at.isoformat() == "2023-01-19T08:30:00+00:00"
+    assert buckets[6].total_count == 15
+
+    assert buckets[7].created_at.isoformat() == "2023-01-19T08:45:00+00:00"
+    assert buckets[7].total_count == 15
+
+
+@pytest.mark.parametrize(
+    "bucket_size, runs_every",
+    [(15, 60), (10, 60), (10, 30), (30, 30), (60, 60), (10, 10), (60, 60 * 4)],
+)
+@pytest.mark.freeze_time("2023-01-19T09:09:47.325132+00:00")
+@pytest.mark.django_db(databases=["analytics"])
+def test_populate_api_usage_bucket(freezer, bucket_size, runs_every):
+    # Given
+    environment_id = 1
+    now = timezone.now()
+    # let's create events at every 1 minutes
+    # for the last two hours, i.e: from 9:09:47
+    for i in range(runs_every * 2):
+        _create_api_usage_event(environment_id, now - timezone.timedelta(minutes=1 * i))
+        # create events in some other environments as well - just to make sure
+        # we don't aggregate them in the same environment
+        _create_api_usage_event(999, now - timezone.timedelta(minutes=1 * i))
+
+    # When
+    populate_api_usage_bucket(bucket_size, run_every=runs_every)
+
+    # Then
+    buckets = (
+        APIUsageBucket.objects.filter(
+            bucket_size=bucket_size,
+            total_count=bucket_size,
+            environment_id=environment_id,
+        )
+        .order_by("-created_at")
+        .all()
+    )
+
+    assert len(buckets) == runs_every // bucket_size
+    # and
+    start_time = timezone.now().replace(minute=0, second=0, microsecond=0)
+    for bucket in buckets:
+        start_time = start_time - timezone.timedelta(minutes=bucket_size)
+        assert bucket.created_at == start_time
+
+
+# @pytest.mark.freeze_time("2023-01-19T09:00:00+00:00")
```

Comment:
Is there a reason this is commented out?

Reply:
Yeah, it was redundant